### PR TITLE
Adding project description code in template intent module

### PR DESCRIPTION
**Pull request by akabhask**
Changes done:
1) Adding project description code in template intent module.
2) Improved pylint score
     ------------------------------------------------------------------
     Your code has been rated at 9.66/10 (previous run: 9.66/10, +0.00)
3) ansible test results
(ansible) bash-3.2$ ansible-test sanity plugins/modules/template_intent.py
Running sanity test "action-plugin-docs"
Running sanity test "ansible-doc"
WARNING: Sanity test "ansible-doc" on Python 3.8 may be slow due to missing libyaml support in PyYAML.
Running sanity test "changelog"
Running sanity test "compile" on Python 2.7
WARNING: Skipping sanity test "compile" on Python 3.5 because it could not be found.
WARNING: Skipping sanity test "compile" on Python 3.6 because it could not be found.
WARNING: Skipping sanity test "compile" on Python 3.7 because it could not be found.
Running sanity test "compile" on Python 3.8
WARNING: Skipping sanity test "compile" on Python 3.9 because it could not be found.
WARNING: Skipping sanity test "compile" on Python 3.10 because it could not be found.
Running sanity test "empty-init"
Running sanity test "future-import-boilerplate"
Running sanity test "ignores"
Running sanity test "import" on Python 2.7
WARNING: Skipping sanity test "import" on Python 3.5 because it could not be found.
WARNING: Skipping sanity test "import" on Python 3.6 because it could not be found.
WARNING: Skipping sanity test "import" on Python 3.7 because it could not be found.
Running sanity test "import" on Python 3.8
WARNING: Skipping sanity test "import" on Python 3.9 because it could not be found.
WARNING: Skipping sanity test "import" on Python 3.10 because it could not be found.
Running sanity test "line-endings"
Running sanity test "metaclass-boilerplate"
Running sanity test "no-assert"
Running sanity test "no-basestring"
Running sanity test "no-dict-iteritems"
Running sanity test "no-dict-iterkeys"
Running sanity test "no-dict-itervalues"
Running sanity test "no-get-exception"
Running sanity test "no-illegal-filenames"
Running sanity test "no-main-display"
Running sanity test "no-smart-quotes"
Running sanity test "no-unicode-literals"
Running sanity test "pep8"
Running sanity test "pslint"
Running sanity test "pylint"
WARNING: Sanity test "pylint" on Python 3.8 may be slow due to missing libyaml support in PyYAML.
Running sanity test "replace-urlopen"
Running sanity test "runtime-metadata"
Running sanity test "shebang"
Running sanity test "shellcheck"
Running sanity test "symlinks"
Running sanity test "use-argspec-type-path"
Running sanity test "use-compat-six"
Running sanity test "validate-modules"
WARNING: Sanity test "validate-modules" on Python 3.8 may be slow due to missing libyaml support in PyYAML.
Running sanity test "yamllint"
WARNING: Skipping sanity test "yamllint" on Python 3.8 due to missing libyaml support in PyYAML.
WARNING: Reviewing previous 14 warning(s):
WARNING: Sanity test "ansible-doc" on Python 3.8 may be slow due to missing libyaml support in PyYAML.
WARNING: Skipping sanity test "compile" on Python 3.5 because it could not be found.
WARNING: Skipping sanity test "compile" on Python 3.6 because it could not be found.
WARNING: Skipping sanity test "compile" on Python 3.7 because it could not be found.
WARNING: Skipping sanity test "compile" on Python 3.9 because it could not be found.
WARNING: Skipping sanity test "compile" on Python 3.10 because it could not be found.
WARNING: Skipping sanity test "import" on Python 3.5 because it could not be found.
WARNING: Skipping sanity test "import" on Python 3.6 because it could not be found.
WARNING: Skipping sanity test "import" on Python 3.7 because it could not be found.
WARNING: Skipping sanity test "import" on Python 3.9 because it could not be found.
WARNING: Skipping sanity test "import" on Python 3.10 because it could not be found.
WARNING: Sanity test "pylint" on Python 3.8 may be slow due to missing libyaml support in PyYAML.
WARNING: Sanity test "validate-modules" on Python 3.8 may be slow due to missing libyaml support in PyYAML.
WARNING: Skipping sanity test "yamllint" on Python 3.8 due to missing libyaml support in PyYAML.

### CODE REVIEW

File: plugins/modules/template_intent.py [hunk 1]
```diff
@@ -602,7 +609,7 @@
 """
 
 RETURN = r"""
-#Case_1: Successful creation/updation/deletion of template
+#Case_1: Successful creation/updation/deletion of template/project
```
Comment (madhansansel) [inline plugins/modules/template_intent.py:612]: Can we leave space after #

File: plugins/modules/template_intent.py [hunk 2]
```diff
@@ -636,7 +643,7 @@
       "msg": String
     }
 
-#Case_3: Given template already exists and requires no udpate
+#Case_3: Given template already exists and requires no update
```
Comment (madhansansel) [inline plugins/modules/template_intent.py:646]: Same as above

File: plugins/modules/template_intent.py [hunk 3]
```diff
@@ -1281,20 +1305,20 @@ def main():
     dnac_template = DnacTemplate(module)
     dnac_template.validate_input().check_return_status("validate_input")
     state = dnac_template.params.get("state")
-
-    for template_config in dnac_template.validated:
+    if state == "merged":
+        api = "get_diff_merged"
```
Comment (madhansansel) [inline plugins/modules/template_intent.py:1309]: instead of api, can we write 
get_diff_state_api = "get_diff_merged"

**Review by madhansansel — APPROVED** [review]
Please check the changes, if required, then update and commit the changes.

File: plugins/modules/template_intent.py (hunk 1 above)
Comment (akabhask) [reply plugins/modules/template_intent.py:612]: Addressed

File: plugins/modules/template_intent.py (hunk 2 above)
Comment (akabhask) [reply plugins/modules/template_intent.py:646]: Addressed

File: plugins/modules/template_intent.py [hunk 4]
```diff
@@ -699,6 +709,11 @@ def check_return_status(self, api_name):
     def validate_input(self):
         """Validate the fields provided in the playbook"""
 
+        if not self.config:
+            self.msg = "config not available for validation"
```
[the remark this answers is not in the document]
Comment (akabhask) [reply plugins/modules/template_intent.py:713]: Addressed

File: plugins/modules/template_intent.py (hunk 3 above)
Comment (akabhask) [reply plugins/modules/template_intent.py:1309]: addressed.

File: plugins/modules/template_intent.py [hunk 5]
```diff
@@ -1281,20 +1305,20 @@ def main():
...
-            dnac_template.get_diff_merge(template_config).check_return_status("get_diff_merge")
-        elif state == "deleted":
-            dnac_template.get_diff_delete(template_config).check_return_status("get_diff_delete")
-        else:
-            dnac_template.status = "invalid"
-            dnac_template.msg = f"State {state} is invalid"
-            dnac_template.check_return_status("main")
+        dnac_template.get_have(config).check_return_status("get_have")
+        dnac_template.get_want(config).check_return_status("get_want")
+        dnac_template.dnac_apply[state](config).check_return_status(api)
```
[the remark this answers is not in the document]
Comment (akabhask) [reply plugins/modules/template_intent.py:1321]: addressed.

**Pull request merged**